vendor/bat-native-ads/src/bat/ads/internal/server/refill_unblinded_tokens/refill_unblinded_tokens.cc

```diff
@@ -53,46 +53,52 @@ void RefillUnblindedTokens::set_delegate(
 }
 
 void RefillUnblindedTokens::MaybeRefill() {
+  if (is_processing_ || retry_timer_.IsRunning()) {
+    return;
+  }
+
+  if (!ShouldRefillUnblindedTokens()) {
+    BLOG(1, "No need to refill unblinded tokens as we already have "
+        << ads_->get_confirmations()->get_unblinded_tokens()->Count()
+            << " unblinded tokens which is above the minimum threshold of "
+                << kMinimumUnblindedTokens);
+    return;
+  }
+
   wallet_ = ads_->get_wallet();
   if (!wallet_.IsValid()) {
     BLOG(0, "Failed to refill unblinded tokens due to an invalid wallet");
     return;
   }
 
-  public_key_ = ads_->get_confirmations()->GetCatalogIssuers().public_key;
-  if (public_key_.empty()) {
-    BLOG(0, "Failed to refill unblinded tokens due to an invalid public key");
+  const CatalogIssuersInfo catalog_issuers =
+      ads_->get_confirmations()->GetCatalogIssuers();
+  if (!catalog_issuers.IsValid()) {
+    BLOG(0, "Failed to refill unblinded tokens due to missing catalog issuers");
     return;
   }
 
+  public_key_ = catalog_issuers.public_key;
```

Comment:
We don't need to check for an empty public key anymore?

Reply:
No because `catalog_issuers.IsValid()` validates the public key. Thanks